File: c/simple_test.cpp
#include <iostream>
#include <string>
#include <stdlib.h>
#include <time.h>

#include "words_to_nums.h"
#include "words_to_nums_fr.h"
#include "words_to_nums_es.h"
//#include "bigint.h"


int main(){

  //words_to_nums converter;
  //words_to_nums_fr converter;
  words_to_nums_es converter;
  /*
  std::string test1= "three hundred thousand six hundred twenty";
  std::cout << converter.converter(test1) << std::endl;

  std::cout << converter.reverse(converter.converter(test1)) << std::endl;
  */

  int test2 = 337936947;
  bigint test2bi = bigint(test2);
  std::cout << converter.reverse(test2) << std::endl;

  int test3 = 110438;
  bigint test2bi2 = bigint(test3);
  std::cout << converter.reverse(test3) << std::endl;
  
  srand(time(NULL));
  int testint;
  bigint value;
  for(int j = 0; j < 100; j++){
    testint = rand();
    value = bigint(testint);
    //std::cout << value << std::endl;
    //std::cout << converter.reverse(value) << std::endl;
    if(value != converter.converter(converter.reverse(value)))
      std::cout << "fail with value: " << value << std::endl;

  }
  

  return 0;

}

File: java/simple_test.java
import java.math.*;
import java.util.LinkedHashMap;
import java.util.Arrays;
import java.util.ArrayList;
import java.util.Collections;
import java.math.BigInteger;
//import java.text.NumberFormat;
//used with bigints

import java.util.Random;

public class simple_test{

    public static void main(String[] args){
	
	//-------------- TEST ---------------------------------
	boolean fail = false;

	
	 
	Random rand = new Random();
	words_to_nums en_test = new words_to_nums();
	//words_to_nums_fr fr_test = new words_to_nums_fr();
	//words_to_nums_es es_test = new words_to_nums_es();
	for(int k= 0; k< 10000;k++){///////////
	    Integer S = rand.nextInt(10000)+1;
	    BigInteger Sbigint = new BigInteger(S.toString());

	    //BigInteger Sbigint = new BigInteger(256/2, rand);

	    //System.out.println(S);
	    
	    String value = en_test.reverse(Sbigint);
	    //String value = fr_test.reverse(Sbigint);
	    //String value = es_test.reverse(Sbigint);

	    BigInteger F = en_test.converter(value);
	    //BigInteger F = fr_test.converter(value);
	    //BigInteger F = es_test.converter(value.trim().replaceAll(" +"," "));

	    if(Sbigint.compareTo(F)!=0)
		fail = true;
	    if(fail){
		System.out.println(Sbigint);
		System.out.println("converter failed with: "+ value);
		System.out.println(F);
		fail = false;
	    }
	    //if(k < 10)
	    //System.out.println(Sbigint);
	}

    }
}
